plugins/Installation/FormDatabaseSetup.php

```diff
@@ -106,11 +106,11 @@ public function createDatabaseObject()
         $port = Adapter::getDefaultPortForAdapter($adapter);
 
         $dbInfos = array(
-            'host'          => $this->getSubmitValue('host'),
-            'username'      => $this->getSubmitValue('username'),
-            'password'      => $this->getSubmitValue('password'),
+            'host'          => trim($this->getSubmitValue('host')),
+            'username'      => trim($this->getSubmitValue('username')),
+            'password'      => trim($this->getSubmitValue('password')),
```

Comment:
A password may end or start with a whitespace maybe? possibly the same for username (although not sure)

Reply:
I'm pretty sure you cannot use whitespaces in MySQL/MariaDB username and password. If anyone can provide source otherwise, you are welcome.

Reply:
just tested it and worked with a whitespace at the end of the password. For username didn't work but not sure if there is a way to make it work...
```sql
CREATE USER 'newuser2'@'%' IDENTIFIED BY 'password ';
GRANT ALL PRIVILEGES ON * . * TO 'newuser2'@'%';
FLUSH PRIVILEGES;
```